### ISSUE TITLE
azure_sd_configs error: cannot list virtual machines: unexpected nextLink host

### ISSUE THREAD

**Issue by zoltan-keller** [VictoriaMetrics/VictoriaMetrics#6912]
### Describe the bug

Issue #6784 is still present in the newest release containing the fix. Azure SD fails with a slightly different error message. 

### To Reproduce

Use the following scrape config:

```
  - job_name: 'azure-sd'
    scheme: https
    tls_config:
      ca_file: /etc/ssl/certs/ca.crt
    follow_redirects: false
    basic_auth:
      username: xxxx
      password: xxxx
    azure_sd_configs:
        - client_id: xxxxx-xxxxx-xxxx
          client_secret: xxxxx-xxxxx-xxxx
          subscription_id: xxxx-xxxx-xxxx
          tenant_id: xxxxx-xxxxx-xxxx
          resource_group: rg-test1
```

### Version

```
~ $ ./vmagent-prod --version
vmagent-20240828-135401-tags-v1.103.0-0-g5aeb759df9
```

### Logs

```
{"ts":"2024-08-29T21:47:15.847Z","level":"error","caller":"VictoriaMetrics/lib/promscrape/config.go:796","msg":"skipping azure_sd_config targets for job_name=azure-sd because of error: cannot list virtual machines: unexpected nextLink host \"management.azure.com:443\", expecting \"management.azure.com\""}
```

### Used command-line flags

```
  httpListenAddr=:8429
  remoteWrite.tlsCAFile="/etc/ssl/certs/ca.crt"
  promscrape.config=/config/scrape.yml
  promscrape.suppressScrapeErrors
  promscrape.suppressDuplicateScrapeTargetErrors
  remoteWrite.showURL
  remoteWrite.tmpDataPath=/tmpData
  remoteWrite.basicAuth.username=xxxx
  remoteWrite.basicAuth.passwordFile=yyyy
  remoteWrite.url=https://xxxxx
  promscrape.maxScrapeSize: "70MiB"
  remoteWrite.maxDiskUsagePerURL: "5GiB"
  envflag.enable: "true"
  envflag.prefix: VM_
  loggerFormat: json
  promscrape.openstackSDCheckInterval: "10m"
  promscrape.httpSDCheckInterval: "10m"
  promscrape.noStaleMarkers: "true"
```

**Comment by jiekun** [comment]
@zoltan-keller Hi. I tried to fix it in the `fix/azure-sd` branch. Would you mind testing it before we could merge? You can [build it from the source code](https://docs.victoriametrics.com/#how-to-build-from-sources). Alternatively, I can provide a binary/docker image if you need it.

**Comment by zoltan-keller** [comment]
Tested it, error message gone and service discovery works now. 

@jiekun

**Comment by hagen1778** [comment]
The bugfix was included into [v1.104.0](https://docs.victoriametrics.com/changelog/#v11040), [v1.102.3](https://docs.victoriametrics.com/changelog/#v11023) and [v1.97.8](https://docs.victoriametrics.com/changelog/#v1978).
